lib/resources/sessions.js

```diff
@@ -48,24 +48,30 @@ module.exports = (service, endpoint) => {
   service.get('/sessions/restore', endpoint((_, { auth }) =>
     auth.session.orElse(Problem.user.notFound())));
 
+  const logOut = (Sessions, auth, session) =>
+    auth.canOrReject('session.end', session.actor)
+      .then(() => Sessions.terminate(session))
+      .then(() => (_, response) => {
+        // revoke the cookie associated w the session, if the session was used to
+        // terminate itself.
```

Comment:
When do you foresee clients invalidating _other_ sessions?  Is there even any point in authorizing a request to invalidate a session?  If a client knows a session token, then isn't that enough authority to invalidate that session?

Reply:
DELETE /v1/sessions/{token} is used for multiple things. One of them is for logging yourself out, but another is revoking access for app users and public links. See the comment above about app users. Project managers can revoke app users' sessions, but someone with a token for an app user or public link shouldn't be able to revoke its access.

On the other hand, DELETE /v1/sessions/current will probably just be used for logout.